### DISCUSSION TITLE
Missing requirement com.ibm.icu.localespi

### DISCUSSION THREAD

**Discussion by rheydenr** [eclipse-tycho/tycho#3781] [Q&A]
Hi all,
I've created a product which contains of multiple plugins. The build runs successfully up to the last step. While materializing, the maven builder says: 

```
Cannot complete the install because one or more required items could not be found.
	Software being installed: MyProduct 2.1.3.202404261009 (MyProduct.ID 2.1.3.202404261009)
	Missing requirement: toolingwin32.win32.x86_64com.ibm.icu.localespi 2.1.3.202404261009 requires 'osgi.bundle; com.ibm.icu.localespi 0.0.0' but it could not be found
	Cannot satisfy dependency:
		From: MyProduct 2.1.3.202404261009 (MyProduct.ID 2.1.3.202404261009)
		To: org.eclipse.equinox.p2.iu; toolingMyProduct.ID.configuration [2.1.3.202404261009,2.1.3.202404261009]
	Cannot satisfy dependency:
		From: toolingMyProduct.ID.configuration 2.1.3.202404261009
		To: org.eclipse.equinox.p2.iu; toolingwin32.win32.x86_64com.ibm.icu.localespi [2.1.3.202404261009,2.1.3.202404261009]
```

I've no idea where to start the search for this error. Any hints are welcome. Thanks in advance, 
Ralf.

**Comment by laeubi** [comment]
Make sure `com.ibm.icu.localespi` is part of your product or remove it from the configuration tab in the product editor.

**Reply by rheydenr** [reply]
You are incredibly fast... Thanks for your answer, that helped me a lot :-) The project compiles now (missing icon problem has to be fixed, but I think I can do this).

**Comment by rheydenr** [comment]
ok, there's one point left. In my current project ("the real one"), I've added persistence.jpa to product configuration and in the product dependencies tab. Despite this, the following error comes when I try to compile the application:

```
[INFO] Resolving dependencies of MavenProject: com.myspace.therealone:com.myspace.therealone.site:2.1.3-SNAPSHOT @ C:\GitHome\TheRealOne-Parent\com.myspace.therealone.site\pom.xml
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: TheRealOne.ID 2.1.3.qualifier
[ERROR]   Missing requirement: TheRealOne.ID 2.1.3.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.persistence.jpa.feature.group 0.0.0' but it could not be found
```

My product file contains:
```
<plugin id="org.eclipse.persistence.jpa"/>
```

The feature file does also contain this plugin:
```
   <plugin
         id="org.eclipse.persistence.jpa"
         version="0.0.0"/>
```
Now is the question: Why does Tycho / Maven complain about a missing dependency?

Thanks in advance,
Ralf.
